Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -1143,4 +1154,55 @@ def problem(text)
       @problems << text
     end
   end
+
+  class TapAuditor
+    attr_reader :name, :path, :tap_audit_exceptions, :problems
+
+    def initialize(tap, options = {})
+      @name                 = tap.name
+      @path                 = tap.path
+      @tap_audit_exceptions = tap.audit_exceptions
+      @strict               = options[:strict]
+      @problems             = []
+    end
+
+    def audit
+      audit_json_files
+      audit_tap_audit_exceptions
+      self
+    end
+
+    def audit_json_files
+      Dir[@path/"**/*.json"].each do |file|
+        JSON.parse Pathname.new(file).read
+      rescue JSON::ParserError
+        problem "#{file.delete_prefix("#{@path}/")} contains invalid JSON"
+      end
+    end
+
+    def audit_tap_audit_exceptions
+      @tap_audit_exceptions.each do |list_name, formula_names|
+        formula_names = formula_names.keys if formula_names.is_a? Hash
+
+        invalid_formulae = []
+        formula_names.each do |name|
+          invalid_formulae.push name if Formulary.factory(name).tap != @name
```

Comment:
```suggestion
          invalid_formulae << name if Formulary[name].tap != @name
```

Reply:
``Error: undefined method `[]' for Formulary:Module``

But I'm fine with using `<<` over `.push` (I don't have a preference)

Reply:
> Edit: ah, you were thinking of `Formula[name]`. Done.

Yup, sorry and thanks!